include/libtorrent/sliding_average.hpp

```diff
@@ -90,8 +92,9 @@ struct average_accumulator
 
 	void add_sample(int s)
 	{
+		TORRENT_ASSERT(s >= 0);
 		++m_num_samples;
-		m_sample_sum += s;
```

Comment:
I wonder if there's a good reason why this only allows positive samples. It seems a bit arbitrary. what would you think about making ``m_sample_sum`` signed instead?